src/plugins/ComputedPlugin.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+
+/**
+ * A plugin for a computed
+ */
+function ComputedPlugin() {
+  const name = 'computed';
+
+  /**
+   * Augment the base schema with the computed type
+   * @param {*} schema
+   */
+  const augmentSchema = ((schema) => {
+    const additions = {
+      inlines: {
+        computed: {
+          nodes: [{
+            match: { object: 'text' }
+          }]
+        },
+      },
+    };
+
+    const newSchema = JSON.parse(JSON.stringify(schema));
+    newSchema.inlines = { ...newSchema.inlines, ...additions.inlines };
+    newSchema.document.nodes[0].match.push({ type: 'computed' });
+    newSchema.blocks.paragraph.nodes[0].match.push({ type: 'computed' });
+    return newSchema;
+  });
+
+  /**
+   * Do not allow computed fields to be edited
+   *
+   * @param {*} value - the Slate value
+   * @param {string} code - the key code
```

Comment:
Is the `code` here used for anything?

Reply:
no, I just left it in so that we know we have access to it if we need it in the future

Reply:
Not in this plugin - but it gives you access to the key that was pressed. Not sure why you'd need that, but it would allow a plugin to disallow certain types of edits.